objstore.go

```diff
@@ -593,12 +603,23 @@ func (b *metricBucket) Upload(ctx context.Context, name string, r io.Reader) err
 	b.ops.WithLabelValues(op).Inc()
 
 	start := time.Now()
-	if err := b.bkt.Upload(ctx, name, r); err != nil {
+	err := b.bkt.Upload(ctx, name, r)
```

Comment:
To be able to count the transferred bytes, you would need to wrap `r`, so you are able get the bytes transferred.

I think you can reuse `newTimingReaderCloser` to wrap this reader here.

Reply:
Agree and will implement it.

Reply:
@simonswine can you please take a look?

Reply:
@ritaCanavarro I think we are getting pretty close, the main concern is that the uploaded bytes do not get recorded yet.

I would expect `objstore_bucket_operation_transferred_bytes_sum{bucket="",operation="upload"} 0` to be `> 0`.


If you look at the line of code above you pass on the unwrapped reader to `r` unmodified to `b.bkt.Upload`, instead you'll need to pass the wrapped reader `trc` to `Upload`, so you are able to count the bytes as they get uploaded.

Reply:
Thanks @simonswine, I fixed it.